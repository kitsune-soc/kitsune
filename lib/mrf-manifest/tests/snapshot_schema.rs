use mrf_manifest::Manifest;

#[test]
fn json_schema() {
    let schema = schemars::schema_for!(Manifest<'_>);
    insta::assert_json_snapshot!(schema);
}
