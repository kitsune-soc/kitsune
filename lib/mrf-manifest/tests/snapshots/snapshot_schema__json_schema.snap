---
source: lib/mrf-manifest/tests/snapshot_schema.rs
expression: schema
---
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "Manifest",
  "description": "Manifest of MRF modules",
  "oneOf": [
    {
      "description": "Manifest v1",
      "type": "object",
      "properties": {
        "activityTypes": {
          "description": "Activity types passed to the MRF module\n\n `*` matching all types",
          "type": "array",
          "items": {
            "type": "string"
          },
          "uniqueItems": true
        },
        "apiVersion": {
          "description": "Version of the MRF API",
          "$ref": "#/$defs/ApiVersion"
        },
        "configSchema": {
          "description": "JSON schema of the configuration passed to the module\n\n This is optional but can be used for automatically generating a configuration UI",
          "anyOf": [
            {
              "$ref": "#/$defs/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "manifestVersion": {
          "type": "string",
          "const": "v1"
        },
        "name": {
          "description": "Name of the MRF module",
          "type": "string"
        },
        "version": {
          "description": "Version of the MRF module",
          "$ref": "#/$defs/SemVer"
        }
      },
      "required": [
        "manifestVersion",
        "apiVersion",
        "name",
        "version",
        "activityTypes"
      ]
    }
  ],
  "$defs": {
    "ApiVersion": {
      "description": "Version of the API used",
      "oneOf": [
        {
          "description": "Version 1",
          "type": "string",
          "const": "v1"
        }
      ]
    },
    "Schema": {
      "type": [
        "object",
        "boolean"
      ]
    },
    "SemVer": {
      "type": "string",
      "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
    }
  }
}
